android/libraries/rib-base/src/androidTest/java/com/badoo/ribs/android/lifecycle/ChildNodeLifecycleTest.kt

```diff
@@ -0,0 +1,107 @@
+package com.badoo.ribs.android.lifecycle
+
+import android.support.test.InstrumentationRegistry
+import android.support.test.annotation.UiThreadTest
+import android.support.test.runner.lifecycle.ActivityLifecycleMonitorRegistry
+import android.support.test.runner.lifecycle.Stage
+import com.badoo.common.ribs.RibsRule
+import com.badoo.ribs.test.util.ribs.root.TestRoot
+import com.badoo.ribs.test.util.ribs.root.TestRootRouter
+import com.badoo.ribs.test.util.waitFor
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import org.junit.Test
+
+
+class ChildNodeLifecycleTest {
+
+    val rootProvider = TestRoot.Provider()
+    val router: TestRootRouter
+        get() = rootProvider.rootNode?.getRouter() as TestRootRouter
+
+    @get:Rule
+    val ribsRule = RibsRule { rootProvider.invoke() }
+
+    @Test
+    fun whenRootIsCreated_noChildrenAreAttached() {
+        assertThat(rootProvider.childNode1?.isAttached).isNull()
+        assertThat(rootProvider.childNode2?.isAttached).isNull()
+    }
+
+    @UiThreadTest
+    @Test
+    fun whenConfigurationIsPushed_child1IsAttached() {
+        router.push(TestRootRouter.Configuration.Node1)
+
+        assertThat(rootProvider.childNode1?.isAttached).isTrue()
+    }
+
+    @UiThreadTest
+    @Test
+    fun whenConfigurationIsPushed_child2IsAttached() {
```

Comment:
Does this test add anything that we don't already know after the previous test at line 33?

Reply:
Done